File: problems/d/detaileddifferences/detaileddifferences.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    n = int(input())
    for _ in range(n):
        first_str = input()
        second_str = input()
        str_size = len(first_str)
        print(first_str)
        print(second_str)
        print("".join(["." if first_str[i] == second_str[i] else "*" for i in range(str_size)]) + "\n")


if __name__ == '__main__':
    main()

File: problems/d/detaileddifferences/detaileddifferences.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;

    cin >> n;
    cin.ignore();

    for (int i=0; i<n; i++) {
        string string_1, string_2;
        cin >> string_1;
        cin.ignore();
        cin >> string_2;
        cin.ignore();
        cout << string_1 << endl;
        cout << string_2 << endl;
        for (size_t i=0; i<string_1.size(); i++) {
            if (string_1[i] == string_2[i]) {
                cout << ".";
            }
            else {
                cout << "*";
            }
        }
        cout << endl << endl;
    }
}